pkg/services/policer/check.go

```diff
@@ -58,74 +61,81 @@ func (p *Policer) processObject(ctx context.Context, addr *addressSDK.Address) {
 		default:
 		}
 
-		p.processNodes(ctx, addr, nn[i], replicas[i].Count())
+		p.processNodes(c, addr, nn[i], replicas[i].Count())
 	}
+
+	if !c.needLocalCopy {
+		p.log.Info("redundant local object copy detected",
+			zap.Stringer("object", addr),
+		)
+
+		p.cbRedundantCopy(addr)
+	}
+}
+
+type processPlacementContext struct {
+	context.Context
+
+	needLocalCopy bool
 }
 
-func (p *Policer) processNodes(ctx context.Context, addr *addressSDK.Address, nodes netmap.Nodes, shortage uint32) {
+func (p *Policer) processNodes(ctx *processPlacementContext, addr *addressSDK.Address, nodes netmap.Nodes, shortage uint32) {
 	log := p.log.With(
 		zap.Stringer("object", addr),
 	)
 
 	prm := new(headsvc.RemoteHeadPrm).WithObjectAddress(addr)
-	redundantLocalCopy := false
 
-	for i := 0; i < len(nodes); i++ {
+	for i := 0; shortage > 0 && i < len(nodes); i++ {
 		select {
 		case <-ctx.Done():
 			return
 		default:
 		}
 
 		if p.netmapKeys.IsLocalKey(nodes[i].PublicKey()) {
-			if shortage == 0 {
-				// we can call the redundant copy callback
-				// here to slightly improve the performance
-				// instead of readability.
-				redundantLocalCopy = true
-				break
-			} else {
-				shortage--
+			if shortage > 0 {
```

Comment:
We have this condition in the loop, how can it ever be false here?

Reply:
Forgot to remove